### PR TITLE
[Semver Verifier] Add support for service declaration changes detection

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR 
- adds semver verifier command support to detect semantic version compatibility on related to service declaration related source code changes.
- adds unit test cases to cover the above mentioned scenarios

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/36082
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/36263

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/util/SyntaxTreeUtils.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+public class SyntaxTreeUtils {
+
+    /**
+     * Returns the node typename of the given {@link SyntaxKind} instance.
+     *
+     * @param nodeKind node kind
+     * @return node typename
+     */
+    public static String getNodeKindName(SyntaxKind nodeKind) {
+        return nodeKind.name().toLowerCase(Locale.ROOT).toLowerCase(Locale.getDefault()).replaceAll("_", " ");
```
Comment (lnash94) [inline misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/util/SyntaxTreeUtils.java:47]: I'm afraid that why we can't use `Locale.ENGLISH` there?

File: misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/util/TestUtils.java [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (lnash94) [inline misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/util/TestUtils.java:1]: Shall we remove this extra new line?

File: misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/FunctionComparator.java [hunk 3]
```diff
@@ -220,13 +219,14 @@ private Optional<? extends Diff> compareReturnTypes(ReturnTypeDescriptorNode new
         return Optional.empty();
     }
 
-    private Optional<NodeDiffImpl<Node>> compareReturnAnnotations(ReturnTypeDescriptorNode newReturn,
-                                                                  ReturnTypeDescriptorNode oldReturn) {
-        NodeList<AnnotationNode> newAnnots = newReturn.annotations();
-        NodeList<AnnotationNode> oldAnnots = oldReturn.annotations();
+    private List<Diff> compareReturnAnnotations(ReturnTypeDescriptorNode newR, ReturnTypeDescriptorNode oldR) {
```
[the remark this answers is not in the document]
Comment (NipunaRanasinghe) [reply misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/FunctionComparator.java]: Renamed as `newReturn` and `oldReturn`

File: misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/util/SyntaxTreeUtils.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/util/SyntaxTreeUtils.java]: Yes `Locale.ENGLISH` would be a safer option for the `toLowerCase` method. Updated the PR.

**Review by lnash94 — APPROVED** [review]
LGTM

**Pull request merged**